subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -166,6 +166,8 @@ public void validateMutation(MutationType type) {
     private boolean dependenciesModified;
     private boolean canBeConsumed = true;
     private boolean canBeResolved = true;
+
+    private boolean canBeMutated = true;
```

Comment:
I see. This is because we have a limited number of mutation types, which are exposed to the user when errors occur. It triggers an error like "Cannot add a dependency after resolution", where the "dependency" term comes from the mutation type. It's more "family of errors".